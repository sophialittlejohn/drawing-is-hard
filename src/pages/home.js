import React from "react";
import { useHistory, useLocation } from "react-router-dom";
import { Container, Grid, Header, Card, Button, Icon } from "semantic-ui-react";

export const Home = () => {
  const { push } = useHistory();
  const { state } = useLocation();

  return (
    <Container style={{ paddingTop: "10%" }} text>
      <Grid>
        <Grid.Row>
          <Grid.Column>
            <Card fluid color="blue">
              <Card.Content>
                <Header as="h1">Drawing Is Hard</Header>
              </Card.Content>
            </Card>
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <Card fluid raised>
              <Card.Content style={{ fontSize: "20px" }}>
                <p>Can a neural network learn to recognize doodling?</p>
                <p>
                  This game is based on Google's{" "}
                  <a href="https://quickdraw.withgoogle.com/">Quick, Draw!</a>{" "}
                  game and uses a sampling it's trained neural network dataset.
                </p>
                <p>
                  It was developed in the context of my capstone project at{" "}
                  <a href="https://www.extensionschool.ch/">
                    EPFL Extension School
                  </a>
                  .
                </p>
                <p>
                  Check out the source code on{" "}
                  <a href="https://github.com/sophialittlejohn/drawing-is-hard">
                    GitHub
                  </a>
                  .
                </p>
              </Card.Content>
            </Card>
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column width="14">
            <Button
              onClick={() => push("/play", state)}
              size="massive"
              primary
              fluid
            >
              Play
            </Button>
          </Grid.Column>
          <Grid.Column width="2">
            <Button
              icon
              size="massive"
              fluid
              basic
              onClick={() => push("/settings")}
            >
              <Icon name="settings" />
            </Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
