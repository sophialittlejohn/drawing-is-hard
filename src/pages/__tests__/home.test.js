import React from "react";
import {
  cleanup,
  fireEvent,
  render,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { Home } from "../home";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

it("checks that play button exists", () => {
  const { getByText } = render(<Home />);
  const playButton = getByText("Play");
  expect(playButton).toBeInTheDocument();

  fireEvent.click(playButton);
  try {
    waitForElementToBeRemoved(() => playButton);
  } catch (er) {
    console.warn(er);
  }
});

it("should take a snapshot", () => {
  const { asFragment } = render(<Home />);

  expect(asFragment(<Home />)).toMatchSnapshot();
});
