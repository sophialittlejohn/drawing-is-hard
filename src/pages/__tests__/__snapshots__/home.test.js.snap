// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui text container"
    style="padding-top: 10%;"
  >
    <div
      class="ui grid"
    >
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui blue fluid card"
          >
            <div
              class="content"
            >
              <h1
                class="ui header"
              >
                Drawing Is Hard
              </h1>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid raised card"
          >
            <div
              class="content"
              style="font-size: 20px;"
            >
              <p>
                This game has been modeled off Google's 
                <a
                  href="https://quickdraw.withgoogle.com/"
                >
                  Quick, Draw!
                </a>
                 game and uses a sampling from the Quick, Draw! dataset.
              </p>
              <p>
                It was developed in the context of my capstone project at 
                <a
                  href="https://www.extensionschool.ch/"
                >
                  EPFL Extension School
                </a>
                .
              </p>
              <p>
                Check out the source code in my 
                <a
                  href="https://github.com/sophialittlejohn/drawing-is-hard"
                >
                  GitHub repo
                </a>
                .
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <button
            class="ui massive fluid primary button"
          >
            Play
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
